src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -55,6 +56,7 @@ public async Task<TimeSpan> RunAsync(CancellationToken cancellationToken)
                 }
                 catch (OperationCanceledException e) when (e.CancellationToken == cancellationToken)
                 {
+                    break;
```

Comment:
Why the added break? This should, presumably, be unnecessary, as the "wait" call below should throw the same exception.

Reply:
Mostly this just seems like a more intentional path? Previously this block was blank, and I tend to prefer not having empty blocks to better make clear my intentions (e.g. Did I forget to implement a catch here?). And at this point the method knows the cancellation has been triggered, so it may as well act rather than let control pass on to a different method assuming it will do the right thing.

Reply:
a "throw" would probably be more appropriate then (that's usually the right answer for "the cancellation exception from the token I'm supposed to be cancelling on").

Reply:
This is an area that's not clear to me. Should cancellations always throw? Or should they acknowledge the cancellation and quickly end. 

It sounds like you're saying to always throw. In that case, it seems like this should not catch OperationCanceledException at all since it has no cleanup to do.

Reply:
That's what most methods do, especially ones that return something (you can't return something if you got cancelled, so the only option is to throw).

I believe it's actually the "correct" behavior.  It will cause your task to enter the "Cancelled" state.  If you quietly return, the Task is "Completed".

It's... frustrating, because it does mean at all levels, you need to _avoid_ catching/swallowing this exception, except at the place that triggered the cancellation in the first place (since he presumably did it to do some other behavior).

Reply:
Aye good explination

Reply:
And I see here it should explicitly throw to avoid having the OperationCanceledException handled by the catch-all Exception below. 

Though I'm not sure why that branch doesn't just also throw. I guess since this is top of the operation stack, restarting the loop is equivalent to restarting the whole thing.

Reply:
There is no real orchestrator.  If this method throws, it will won't run again (at least, not correctly).  This _is_ the operational loop. It has some odd timing needs (to deal with Kusto ingestion lag), which I didn't have a good way to communicate outward, so this guy just sits in a loop dealing with those.